packages/editor/src/components/rich-text/index.js

```diff
@@ -450,9 +452,12 @@ export class RichText extends Component {
 			this.applyRecord( record );
 		}
 
-		const { start, end } = record;
+		// Filter out annotations
+		const filteredRecord = removeFormat( record, 'core/annotation', 0, record.text.length );
```

Comment:
I'm working on a hook/api to prepare values for conversion to an editable tree. This will not only be useful to annotations, but also to spell checkers, visualisation of invisible characters etc. We cannot hardcode `core/annotation` in the rich text component.

Reply:
Would be good to remove `removeFormat` down inside `this.valueToFormat`.